ctapipe/io/containers.py

```diff
@@ -78,6 +77,7 @@ class RawCameraContainer(Container):
                             "integrated ADC data (n_pixels)"))
     adc_samples = Item(Map(), ("map of channel to arrays of "
                                "(n_pixels, n_samples)"))
+    num_samples = Item(int, "number of time samples")
```

Comment:
The reason it changes per event is because its obtained from `hsdata->event.teldata[itel].raw->num_samples`. This used to be where num_channels was obtained from before my pyhessio PR, but now i obtain it from `hsdata->camera_org[itel].num_gains`, which allows you to obtain the num channels for every telescope, even if the telescope has no data in the current event.

Sure num_channels will be dropped, but its num_samples that we have a problem about in this discussion.

Would you be happier with solution I used in 0a9e02d33187741be71b49ca41dace75333de2af, where i fill the num_samples[tel_id] as soon as we reach an event where the tel_id has data. This is similar to the method used in master, however it is only num_samples that updates at each new event, instead of the whole inst container.